OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs

```diff
@@ -86,12 +86,28 @@ public void PlayMusic(string track = null, LuaFunction func = null)
 				playlist.Play(musicInfo);
 		}
 
+		[Desc("Play track defined in music.yaml as background music.")]
+		public void PlayBackgroundMusic(string track)
+		{
+			if (!playlist.IsMusicAvailable)
+				return;
+
+			playlist.Play(world.Map.Rules.Music[track]);
```

Comment:
I don't think this will work.  You need to set the `currentBackgroundSong` variable in MusicPlaylist in order to get the right behaviour.  I suggest adding a new SetBackgroundMusic method to MusicPlaylist to set that and CurrentSongAsBackground (and then change that back to private set).